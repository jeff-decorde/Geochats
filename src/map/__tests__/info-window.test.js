import React from 'react';
import LocationInfoWindow from '../info-window.js';
import markerMock from '../../mocks/marker.js';

describe('LocationInfoWindow', () => {
  const wrapper = shallow(
    <LocationInfoWindow
      marker={{
        data: markerMock
      }}
    />
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot(wrapper);
  });

  it('should pass the right props', () => {
    expect(wrapper.find('InfoWindow').length).toEqual(1);
    const spans = wrapper.find('span');
    expect(spans.at(0).text()).toEqual(markerMock.name);
    expect(spans.at(1).text()).toEqual(`${markerMock.radius}`);
    expect(spans.at(2).text()).toEqual(`${markerMock.maxRadius}`);
    expect(spans.at(3).text()).toEqual(`${markerMock.coord[0].latitude}, ${markerMock.coord[0].longitude}`);
  })
});
