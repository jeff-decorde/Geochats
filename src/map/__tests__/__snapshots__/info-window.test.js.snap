// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<InfoWindow>
  <div>
    <div>
      <b>
        Name : 
      </b>
      <span>
        name
      </span>
    </div>
    <div>
      <b>
        Radius : 
      </b>
      <span>
        1000
      </span>
    </div>
    <div>
      <b>
        Maximum radius : 
      </b>
      <span>
        10000
      </span>
    </div>
    <div>
      <b>
        Coordinates : 
      </b>
      <span>
        12345, 67890
      </span>
    </div>
  </div>
</InfoWindow>
`;
